### ISSUE TITLE
Add endpoint to get a fresh PID without creating a dataset

### ISSUE THREAD

**Issue by jl-wynen** [SciCatProject/scicat-backend-next#314]
When uploading datasets, Scitacean first generates a dataset PID locally before uploading to SciCat. It does this because it uses the id to determine where to place the files. And it uploads the files before the dataset because a file upload can be reverted more easily than an upload to SciCat.

This approach can potentially lead to the dataset being rejected because the id is already in use. So it would be useful to have an endpoint that returns a new, unused id but doesn't actually create a dataset in the database.

If feasible, it would be extra nice, to avoid race conditions, to also reserve the id. But then it may need to be freed up if the dataset is not actually uploaded, e.g. after 1 day or so.

This is mainly useful for datasets. I personally see no need to support it with other models as well.

**Comment by nitrosx** [comment]
I agree with you that from the user perspective the dataset creation (aka file upload, dataset and origdatablock creation) should be an atomic operation and been able to roll it back should be handled automatically by the system.

I wonder if we could propose something like this:
1. Request/Open a transaction which returns a transaction id and a dataset pid (which is reserved)
2. Upload the files using the dataset pid provided
3. Create the dataset referencing the transaction id which assign the correct dataset pid
4. Create the origdatablock referencing the transaction id which assign the correct dataset pid
5. Close successfully the transaction and everybody is happy (Now dataset cannot be deleted)
6. If anything goes wrong, user rolls back the transaction and the system remove dataset and origdatablock assigned.
7. User is responsible to delete files from server

If you think that something like this can work, we can propose it and implement it

**Comment by jl-wynen** [comment]
I think this would work from a user perspective.

**Comment by nitrosx** [comment]
If transactions are still relevant, I will put the on the agenda for on of the following meetings.

**Comment by jl-wynen** [comment]
A bit late now, but transactions are still relevant.

The precise steps and their order from your earlier message should probably be adjusted. It would be good to upload dataset and datablock first to ensure that they are valid and fail before the costly file upload. But this would potentially require amending them after the file upload.
It is a little too complicated to discuss in text here. We should discuss in person.
